pkg/events/queue/queue.go

```diff
@@ -0,0 +1,33 @@
+package queue
+
+import (
+	"github.com/aquasecurity/tracee/pkg/external"
+)
+
+type CacheType int
+
+const (
+	EventQueueTypeNone CacheType = iota
+	EventQueueTypeMem
+)
+
+type CacheConfig struct {
+	EventsCache        bool
+	EventsCacheType    CacheType
```

Comment:
Okay so.. instead of having a config struct, we now configure the EventQueue implementation directly based on its type. AND there is Setup() function does not require any argument (check flags-cache.go). Also, I'm implementing a String() function to describe the cache type being loaded in debug.

Apart from that, I **think** that I implemented what you suggested, let me know if not, please. Thanks.